src/rt-app.c

```diff
@@ -759,6 +772,15 @@ void *thread_body(void *arg)
 				NULL);
 	}
 
+	/* TODO find a better way to handle that constraint:
+	 * Set the task to SCHED_DEADLINE as far as possible touching its
+	 * budget as little as possible for the first iteration.
+	 */
+
+	/* Set scheduling policy and print pretty info on stdout */
+	log_notice("[%d] Starting with %s policy with priority %d", data->ind, policy_to_string(data->sched_data->policy), data->sched_data->prio);
+	set_thread_priority(data, data->sched_data);
```

Comment:
I'm afraid that this is not enough with config:
```
"threadA" : {
    "priority" : 0,
    "run" : 5000,
    "timer" :  { "ref" : "tickB", "period" : 7777, }
}
```
Because sched_data is not null but doesn't have default policy.

The change below should fix the problem:
```
@ -852,6 +852,14 @@ parse_thread_data(char *name, struct json_object *obj, int index,
 		data->phases = malloc(sizeof(phase_data_t) * data->nphases);
 		parse_thread_phase_data(obj,  &data->phases[0], opts, (long)data);
 
+		/* There is no "phases" object which means that thread and phase will
+		 * use same scheduling parameters. But thread object looks for default
+		 * value when parameters are not defined whereas phase doesn't.
+		 * We remove phase's scheduling policy which is a subset of thread's one
+		 */
+		free(data->phases[0].sched_data);
+		data->phases[0].sched_data = NULL;
+
 		/*
 		 * Get loop number:
 		 *
```